Classes/Plugin/Feeds.php

```diff
@@ -88,28 +88,55 @@ public function main($content, $conf)
             $additionalWhere = '';
             // Check for pre-selected collections.
             if (!empty($this->piVars['collection'])) {
-                $additionalWhere = ' AND tx_dlf_collections.uid=' . intval($this->piVars['collection']);
+                $additionalWhere = 'tx_dlf_collections.uid=' . intval($this->piVars['collection']);
             } elseif (!empty($this->conf['collections'])) {
-                $additionalWhere = ' AND tx_dlf_collections.uid IN (' . $GLOBALS['TYPO3_DB']->cleanIntList($this->conf['collections']) . ')';
+                $additionalWhere = 'tx_dlf_collections.uid IN (' . $GLOBALS['TYPO3_DB']->cleanIntList($this->conf['collections']) . ')';
             }
-            $result = $GLOBALS['TYPO3_DB']->exec_SELECT_mm_query(
-                'tx_dlf_documents.uid AS uid,tx_dlf_documents.partof AS partof,tx_dlf_documents.title AS title,tx_dlf_documents.volume AS volume,tx_dlf_documents.author AS author,tx_dlf_documents.record_id AS guid,tx_dlf_documents.tstamp AS tstamp,tx_dlf_documents.crdate AS crdate',
-                'tx_dlf_documents',
-                'tx_dlf_relations',
-                'tx_dlf_collections',
-                'AND tx_dlf_documents.pid=' . intval($this->conf['pages'])
-                    . ' AND tx_dlf_relations.ident=' . $GLOBALS['TYPO3_DB']->fullQuoteStr('docs_colls', 'tx_dlf_relations')
-                    . ' AND tx_dlf_collections.pid=' . intval($this->conf['pages'])
-                    . $additionalWhere
-                    . Helper::whereClause('tx_dlf_documents')
-                    . Helper::whereClause('tx_dlf_collections'),
-                'tx_dlf_documents.uid',
-                'tx_dlf_documents.tstamp DESC',
-                intval($this->conf['limit'])
-            );
-            if ($GLOBALS['TYPO3_DB']->sql_num_rows($result)) {
+
+            $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+                ->getQueryBuilderForTable('tx_dlf_documents');
+
+            $result = $queryBuilder
+                ->select(
+                    'tx_dlf_documents.uid AS uid',
+                    'tx_dlf_documents.partof AS partof',
+                    'tx_dlf_documents.title AS title',
+                    'tx_dlf_documents.volume AS volume',
+                    'tx_dlf_documents.author AS author',
+                    'tx_dlf_documents.record_id AS guid',
+                    'tx_dlf_documents.tstamp AS tstamp',
+                    'tx_dlf_documents.crdate AS crdate'
+                )
+                ->from('tx_dlf_documents')
+                ->join(
+                    'tx_dlf_documents',
+                    'tx_dlf_relations',
+                    'tx_dlf_relations',
```

Comment:
Shouldn't this be something else like `tx_dlf_relations_join`?

Reply:
The third parameter is only an identifier. You can call it what you want. If you dont want to use an identifier, you can take the table name.

Reply:
Isn't this a problem since we are using the identifier later in the `WHERE` clauses? How does Doctrine know we are referring to the joined table instead of the original?

Reply:
I can rename it. Thats not the problem. But I think the better identifier is "tx_dlf_documents_collections_mm".